### PR TITLE
credentials manager initial poc

### PR DESCRIPTION
**Pull request by johnnyon-amzn**
###Description
Allows users to create multiple login profiles and access keys.
Added a pop up window for handling credentials and saving new profiles.
Created an edit button that also updates config.
Changed config.json file to be able to store credentials.

###Issues resolved
https://github.com/opensearch-project/dashboards-desktop/issues/8

### CODE REVIEW

File: src/renderer.js [hunk 1]
```diff
@@ -6,41 +7,54 @@ function getMapping() {
...
+        return `<option value="${name}">${name}</option>`
+    });
+    profile.innerHTML += options.join('');
+
+}
+
+function getActiveConfig() {
+    let profile = document.getElementById("profile");
+    var name = profile.options[profile.selectedIndex].value;
+    console.log('name', name);
```
Comment (zhongnansu) [inline src/renderer.js:29]: could you check similar console.log, and see if they are really necessary to remove. If it's for dev purpose, you can add some TODO comments, so people who work on this later will know and remove.

File: src/renderer.js [hunk 2]
```diff
@@ -6,41 +7,54 @@ function getMapping() {
...
 
 async function populateFields() {
-    let config = window.config;
-    let mapping = getMapping();
-    for (map of mapping) {
-        let input = document.getElementById(map.id);
-        input.setAttribute("value", config[map.key])
-    }
+    config = window.electron.getConfig();
+    let profile = document.getElementById("profile");
```
Comment (zhongnansu) [inline src/renderer.js:16]: Np: you may want to make id as constants for reuse. See another usage at line 27

**Review by zhongnansu — COMMENTED** [review]
Please add license headers for all code files. You may consider some utility tools in VScode that can automatically add header for all files in the project.

File: src/config.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+const path = require('path');
+const fs = require("fs");
+const CONFIG_PATH = path.join(__dirname, "config.json");
```
[the remark this answers is not in the document]
Comment (johnnyon-amzn) [reply src/config.js:3]: Not sure how to approach. By default it should be a file in src that is an empty object. However, when testing and using the app, it gets populated with access key information which I don't want to push

[the remark this answers is not in the document]
Comment (zhongnansu) [reply src/config.js:21]: OK so this file will be generated by code? I am good with that. Any reference in README or comments in code files will be helpful

**Review by zhongnansu — APPROVED** [review]

**Pull request merged**